pandas/core/arrays/numpy_.py

```diff
@@ -0,0 +1,431 @@
+from functools import wraps
+import numbers
+
+import numpy as np
+
+from pandas._libs import lib
+
+from pandas.core.dtypes.common import extract_array
+from pandas.core.dtypes.dtypes import ExtensionDtype
+from pandas.core.dtypes.generic import ABCIndexClass, ABCSeries
+from pandas.core.dtypes.inference import is_list_like
+
+from pandas import compat
+from pandas.core import nanops
+
+from .base import ExtensionArray, ExtensionOpsMixin
+
+
+def _validate_reduction_args(func):
+    """
+    Validate that unused keyword arguments are not set.
+    """
+    check = {'out', 'keepdims', 'initial', 'dtype'}
+
+    msg = (
+        "'{}' does not support the '{}' parameter."
+    )
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if len(args) > 1:
+            raise TypeError(
+                "'{}' requires keyword arguments.".format(func.__name__)
+            )
+        for kw, value in kwargs.items():
+            if kw in check and value is not None:
+                raise TypeError(msg.format(func.__name__, kw))
+            elif kw == 'ovewrite_input' and kw is not False:
+                raise TypeError(msg.format(func.__name__, kw))
+
+        return func(*args, **kwargs)
+    return wrapper
+
+
+class PandasDtype(ExtensionDtype):
+    """
+    A Pandas ExtensionDtype for NumPy dtypes.
+
+    .. versionadded:: 0.24.0
+
+    This is mostly for internal compatibility, and is not especially
+    useful on its own.
+
+    Parameters
+    ----------
+    dtype : numpy.dtype
+    """
+    _metadata = ('_dtype',)
+
+    def __init__(self, dtype):
+        dtype = np.dtype(dtype)
+        self._dtype = dtype
+        self._name = dtype.name
+        self._type = dtype.type
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def type(self):
+        return self._type
+
+    @property
+    def _is_numeric(self):
+        # exclude object, str, unicode, void.
+        return self.kind in set('biufcmM')
```

Comment:
why include datetime64 and timedelta64?  These seem like footguns.  Or is it for internal consistency with [...]?

Reply:
Removed. I didn't release those blocks set `is_numeric` to False.